src/Microsoft.OpenApi/Services/OpenApiFilterService.cs

```diff
@@ -46,16 +56,52 @@ public static class OpenApiFilterService
                 {
                     var regex = new Regex(tagsArray[0]);
 
-                    predicate = (o) => o.Tags.Any(t => regex.IsMatch(t.Name));
+                    predicate = (url, operationType, operation) => operation.Tags.Any(tag => regex.IsMatch(tag.Name));
                 }
                 else
                 {
-                    predicate = (o) => o.Tags.Any(t => tagsArray.Contains(t.Name));
+                    predicate = (url, operationType, operation) => operation.Tags.Any(tag => tagsArray.Contains(tag.Name));
+                }
+            }
+            else if (requestUrls != null)
+            {
+                List<string> operationTypes = new List<string>();
+
+                var graphVersion = source.Info.Version;
```

Comment:
[Nit]: Consider renaming to `apiVersion` so as to be generic and accommodate other non-Graph APIs that may be used here.

Reply:
Yeah this is a good point actually. Thanks for pointing it out.